src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryTable.cs

```diff
@@ -15,7 +15,105 @@
 namespace Microsoft.PowerFx.Functions
 {
     internal static partial class Library
-    {
+    {
+        public static async ValueTask<FormulaValue> Collect(EvalVisitor runner, EvalVisitorContext context, IRContext irContext, FormulaValue[] args)
+        {
+            FormulaValue arg0;
+            var argc = args.Length;
+            var returnIsTable = args.Length > 2;
+
+            // Need to check if the Lazy first argument has been evaluated since it may have already been
+            // evaluated in the ClearCollect case.
+            if (args[0] is LambdaFormulaValue arg0lazy)
+            {
+                arg0 = await arg0lazy.EvalAsync().ConfigureAwait(false);
+            }
+            else
+            {
+                arg0 = args[0];
+            }
+
+            if (arg0 is BlankValue)
+            {
+                return arg0;
+            }
+            else if (arg0 is ErrorValue)
```

Comment:
>else  [](http://example.com/codeflow?start=12&length=5)

Here it is using `else if` following a return statement, but on line 45 there is no `else`. We should be consistent (e.g., remove this `else` from here)